src/compiler/checker.ts

```diff
@@ -37954,9 +37955,16 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             case SyntaxKind.JsxAttributes:
                 return checkJsxAttributes(node as JsxAttributes, checkMode);
             case SyntaxKind.JsxOpeningElement:
-                Debug.fail("Shouldn't ever directly check a JsxOpeningElement");
+                return Debug.fail("Shouldn't ever directly check a JsxOpeningElement");
+            case SyntaxKind.ImportKeyword:
+                return isImportCall(node.parent) || isImportMeta(node.parent) ? anyType : errorType; // LHS of ImportCall or ImportMeta or invalid expression TODO: Should this have a better type?
+            case SyntaxKind.JsxNamespacedName:
+                return errorType; // FIXME: Remove when JsxNamespacedName is no longer used as an expression
+            case SyntaxKind.MissingDeclaration:
+                return errorType; // invalid decorated expression

```

Comment:
@rbuckton should this be an expression? It's not flagged by `isExpressionNode`, but is definitely assigned to expression-expecting node positions.

Reply:
We have two different ways to test for expressions, so we may need to verify both are correct:

- `isExpression` in _utilitiesPublic.ts_ &mdash; Tests whether a `Node` is a subtype of `Expression`. Doesn't care about context (i.e., no `.parent` access). Designed for use with transformers (where `.parent` may not be set).
- `isExpressionNode` in __utilities.ts_ &mdash; Tests whether a `Node` is used in an expression position. Highly contextual (performs `.parent` lookups). Used by the checker, though I'd love to find a way to retire it at some point.